pyDF-data/libdfdata/torch_dataloader.py

```diff
@@ -161,6 +161,15 @@ def cleanup_pin_memory_thread(self):
                         _ = self.data_queue.get(timeout=2)
                     except queue.Empty:
                         break
+                if hasattr(self.data_queue, "task_done"):
+                    while True:
+                        # self.data_queue.get() without a subsequent task_done()
+                        # will cause self.data_queue.join() to wait indefinitely.
+                        try:
+                            self.data_queue.task_done()
+                        except ValueError:
```

Comment:
In which cases does this ValueError occur? Do we need this exception handling?

Reply:
I started with a simpler solution that doesn't rely on ValueError.  As long as it can be guaranteed that the data loader is the only consumer of the data_queue I can remove the exception handling.

Reply:
I am fine with the except block, if it is documented why we need this, or when such an exception may happen.

Reply:
I removed the except block. To be honest, this is a better approach. The two places where items are removed from the queue in this code they're followed w/ task_done() calls. If, for some reason, there are other pending tasks it is better to detect them (with a hang) than blindly clearing them.